src/http/fetch_request.js

```diff
@@ -124,7 +124,7 @@ export class FetchRequest {
     try {
       this.delegate.requestStarted(this)
       const response = await fetch(this.url.href, fetchOptions)
-      return await this.receive(response)
```

Comment:
Should this `await` remain so that the `finally` block calling `this.delegate.requestFinished(this)` is deferred until the response has been processed/rendered?

As it stands on main branch, and with your changes too, it gets called prior to the response being downloaded/processed.

Reply:
Great catch! I've reverted that.